packages/component/src/providers/ModalDialog/private/Popover.tsx

```diff
@@ -0,0 +1,84 @@
+import { hooks } from 'botframework-webchat-api';
+import { useRefFrom } from 'use-ref-from';
+import classNames from 'classnames';
+import React, { memo, type PropsWithChildren, useCallback, useEffect, useRef } from 'react';
+
+import useStyleSet from '../../../hooks/useStyleSet';
+
+const { useLocalizer } = hooks;
+
+type Props = PropsWithChildren<{
+  // "defaultProps" is being deprecated.
+  // eslint-disable-next-line react/require-default-props
+  'aria-label'?: string;
+  // "defaultProps" is being deprecated.
+  // eslint-disable-next-line react/require-default-props
+  'aria-labelledby'?: string;
+  // "defaultProps" is being deprecated.
+  // eslint-disable-next-line react/require-default-props
+  className?: string;
+  // "defaultProps" is being deprecated.
+  // eslint-disable-next-line react/require-default-props
+  onDismiss?: () => void;
+}>;
+
+const ModalDialog = memo(
+  ({ 'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledBy, children, className, onDismiss }: Props) => {
+    const [{ modalDialog: modalDialogStyleSet }] = useStyleSet();
+    const dialogRef = useRef<HTMLDialogElement>(null);
+    const localize = useLocalizer();
+    const onDismissRef = useRefFrom(onDismiss);
+
+    const closeButtonAlt = localize('KEYBOARD_HELP_CLOSE_BUTTON_ALT');
+
+    const handleCloseButtonClick = useCallback(() => dialogRef.current?.close(), [dialogRef]);
+    const handleClose = useCallback(() => onDismissRef.current?.(), [onDismissRef]);
+
+    useEffect(() => dialogRef.current?.showModal(), [dialogRef]);
+
+    return (
+      <dialog
+        // When "aria-labelledby" is set, it must not set "aria-label".
+        aria-label={!ariaLabelledBy ? ariaLabel : undefined}
+        aria-labelledby={ariaLabelledBy}
+        className={classNames('webchat__modal-dialog', className, modalDialogStyleSet + '')}
+        onClose={handleClose}
```

Comment:
I don' think you need to wrap `onDismissRef.current` in a callback.
```suggestion
        onClose={onDismissRef.current}
```